lib/_stream_readable.js

```diff
@@ -1040,11 +1040,6 @@ function copyFromBuffer(n, list) {
 function endReadable(stream) {
   var state = stream._readableState;
 
-  // If we get here before consuming all the bytes, then that is a
-  // bug in node.  Should never happen.
-  if (state.length > 0)
-    throw new Error('"endReadable()" called on non-empty stream');
```

Comment:
I would keep this safety check in.

Reply:
It is dead code at the moment. There is no way to trigger it right now and only a false refactoring could trigger this.

Reply:
Just as reference - there are three occurrences of `endReadable` right now. Each of those are only triggered in case `state.length === 0`. Those guards are here
https://github.com/BridgeAR/node/blob/615ab68c02e81bc17dbfb15d489a3558f667dc67/lib/_stream_readable.js#L384
https://github.com/BridgeAR/node/blob/615ab68c02e81bc17dbfb15d489a3558f667dc67/lib/_stream_readable.js#L395
https://github.com/BridgeAR/node/blob/615ab68c02e81bc17dbfb15d489a3558f667dc67/lib/_stream_readable.js#L466
The function can also not be triggered from the outside.

Reply:
This can be removed then, 👍 .